client/asset/eth/eth.go

```diff
@@ -0,0 +1,534 @@
+//
+// This method blocks until the redemption is found, an error occurs or the
+// provided context is canceled.
+func (*ExchangeWallet) FindRedemption(ctx context.Context, coinID dex.Bytes) (redemptionCoin, secret dex.Bytes, err error) {
+	return nil, nil, notImplementedErr
+}
+
+// Refund refunds a contract. This can only be used after the time lock has
+// expired.
+// NOTE: The contract cannot be retrieved from the unspent coin info as the
+// wallet does not store it, even though it was known when the init transaction
+// was created. The client should store this information for persistence across
+// sessions.
+func (*ExchangeWallet) Refund(coinID, contract dex.Bytes) (dex.Bytes, error) {
+	return nil, notImplementedErr
+}
+
+// Address returns an address for the exchange wallet.
+func (eth *ExchangeWallet) Address() (string, error) {
+	eth.acctMtx.RLock()
+	defer eth.acctMtx.RUnlock()
+	return eth.acct.Address.String(), nil
+}
+
+// Unlock unlocks the exchange wallet.
+func (eth *ExchangeWallet) Unlock(pw string) error {
+	eth.acctMtx.RLock()
+	defer eth.acctMtx.RUnlock()
+	return eth.node.unlock(eth.ctx, pw, eth.acct)
+}
+
+// Lock locks the exchange wallet.
+func (eth *ExchangeWallet) Lock() error {
+	eth.acctMtx.RLock()
+	defer eth.acctMtx.RUnlock()
+	return eth.node.lock(eth.ctx, eth.acct)
+}
+
+// Locked will be true if the wallet is currently locked.
+func (eth *ExchangeWallet) Locked() bool {
+	wallets, err := eth.node.listWallets(eth.ctx)
+	if err != nil {
+		eth.log.Errorf("list wallets error: %v", err)
+		return false
+	}
+	eth.acctMtx.RLock()
+	defer eth.acctMtx.RUnlock()
+	var wallet rawWallet
+	findWallet := func() bool {
+		for _, w := range wallets {
+			for _, a := range w.Accounts {
+				if bytes.Equal(a.Address[:], eth.acct.Address[:]) {
+					wallet = w
+					return true
+				}
+			}
+		}
+		return false
+	}
+	if !findWallet() {
+		eth.log.Error("unable to find wallet for account: %v", eth.acct)
+		return false
+	}
+	return wallet.Status != "Unlocked"
+}
+
+// PayFee sends the dex registration fee. Transaction fees are in addition to
+// the registration fee, and the fee rate is taken from the DEX configuration.
+//
+// NOTE: PayFee is not intended to be used with Ethereum at this time.
+func (*ExchangeWallet) PayFee(address string, regFee uint64) (asset.Coin, error) {
+	return nil, notImplementedErr
+}
+
+// sendToAddr sends funds from acct to addr.
+func (eth *ExchangeWallet) sendToAddr(addr common.Address, amt, gasFee *big.Int) (common.Hash, error) {
+	tx := map[string]string{
+		"from":     fmt.Sprintf("0x%x", eth.acct.Address),
+		"to":       fmt.Sprintf("0x%x", addr),
+		"value":    fmt.Sprintf("0x%x", amt),
+		"gasPrice": fmt.Sprintf("0x%x", gasFee),
+	}
+	return eth.node.sendTransaction(eth.ctx, tx)
+}
+
+// Withdraw withdraws funds to the specified address. Fees are subtracted from
+// the value.
+//
+// TODO: Value could be larger than a uint64. Deal with it...
+// TODO: Return the asset.Coin.
```

Comment:
Assuming the return is a TODO because coin ID needs adaptation to support general txids (no contract business)? i.e. "withdrawals" in https://github.com/decred/dcrdex/issues/1021

Reply:
Yes, it is not set in stone what the return should look like yet, with flags.